pkgs/development/python-modules/opentelemetry-instrumentation-threading/default.nix

```
@@ -0,0 +1,46 @@
+{
+  buildPythonPackage,
+  pythonOlder,
+
+  # build-system
+  hatchling,
+
+  # dependencies
+  opentelemetry-api,
+  opentelemetry-instrumentation,
+  wrapt,
+
+  # tests
+  opentelemetry-test-utils,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  inherit (opentelemetry-instrumentation) version src;
+  pname = "opentelemetry-instrumentation-threading";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  sourceRoot = "${opentelemetry-instrumentation.src.name}/instrumentation/${pname}";
```

```suggestion
  sourceRoot = "${opentelemetry-instrumentation.src.name}/instrumentation/opentelemetry-instrumentation-threading";
```
Please don't reuse `pname` here since overriding the pname will break the package.